lib/hanami/cli/generators/app/relation.rb

```diff
@@ -19,8 +19,14 @@ def initialize(fs:, inflector:, out: $stdout)
 
           # @since 2.2.0
           # @api private
-          def call(key:, namespace:, base_path:)
+          def call(key:, namespace:, base_path:, gateway:)
             schema_name = key.split(KEY_SEPARATOR).last
+            body_content = ["schema :#{schema_name}, infer: true"]
+
+            if gateway
+              gateway_name = gateway.split(KEY_SEPARATOR).last
+              body_content.unshift("gateway :#{gateway_name}")
```

Comment:
```suggestion
              body_content.unshift("gateway :#{gateway}")
```

I don't think there's any circumstance in which gateways would have complex names with delimiters or separators, so I think this simpler approach will do just fine :)

Reply:
Could we also use `.prepend` instead of `.unshift`? "Unshift" as a name truly breaks my brain, and I reckon I mustn't be the only one!